### PR TITLE
Enable triggering postcommit via specific file

### PR DESCRIPTION
**Pull request by Abacn**
Workaround #28909 

Enable to trigger a PostCommit from pull request by touching a file named "<workflow_name>.json" (/json to get around with RAT PreCommit), as a simple workaround of #28909.

In this way the last piece of gap (comment trigger) is partially fixed and we can shutdown all Jenkins jobs.

**Please** add a meaningful description for your change here

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: .github/workflows/README.md [hunk 1]
```diff
@@ -271,7 +271,6 @@ Please note that jobs with matrix need to have matrix element in the comment. Ex
 
 | Workflow name | Matrix | Trigger Phrase | Cron Status |
 |:-------------:|:------:|:--------------:|:-----------:|
-| [ PostCommit BeamMetrics Publish ](https://github.com/apache/beam/actions/workflows/beam_PostCommit_BeamMetrics_Publish.yml) | N/A |`Run Beam Metrics Deployment`| [![.github/workflows/beam_PostCommit_BeamMetrics_Publish.yml](https://github.com/apache/beam/actions/workflows/beam_PostCommit_BeamMetrics_Publish.yml/badge.svg?event=schedule)](https://github.com/apache/beam/actions/workflows/beam_PostCommit_BeamMetrics_Publish.yml?query=event%3Aschedule)
```
Comment (Abacn) [inline .github/workflows/README.md:274]: BeamMetrics Publish and Website Publish are not really "postcommit" or test suites. It **deploys** the Beam website or **deploys** the community metrics dashboard. They should not have comment trigger (which could be used to attack beam website). So renamed them and move to "other" category.

File: .github/workflows/README.md [hunk 2]
```diff
@@ -271,7 +271,6 @@ Please note that jobs with matrix need to have matrix element in the comment. Ex
...
 | [ PostCommit Go ](https://github.com/apache/beam/actions/workflows/beam_PostCommit_Go.yml) | N/A |`Run Go PostCommit`| [![.github/workflows/beam_PostCommit_Go.yml](https://github.com/apache/beam/actions/workflows/beam_PostCommit_Go.yml/badge.svg?event=schedule)](https://github.com/apache/beam/actions/workflows/beam_PostCommit_Go.yml?query=event%3Aschedule) |
```
Comment (damccorm) [inline .github/workflows/README.md:274]: We need to also update all the postcommits in this file to not have trigger phrases listed. Ideally we'd list the file to modify instead.

We should also modify the part that talks about trigger phrases at the top and explain this approach.

File: .github/workflows/beam_Java_JMH.yml [hunk 3]
```diff
@@ -19,7 +19,7 @@ on:
   schedule:
     - cron: '0 0 * * 0'
   pull_request_target:
-    paths: ['release/trigger_all_tests.json']
+    paths: ['release/trigger_all_tests.json', 'beam_Java_JMH.json']
```
Comment (damccorm) [inline .github/workflows/beam_Java_JMH.yml:22]: Can we put these in a subfolder? Maybe `.github/trigger_files`? That way if they accidentally get checked in it doesn't cause issues.

Comment (damccorm) [reply .github/workflows/beam_Java_JMH.yml:22]: Our guidance could even be to check them in, it does no harm.

**Review by damccorm — COMMENTED** [review]
I'm on board with this approach, thanks

File: .github/workflows/beam_Java_JMH.yml (hunk 3 above)
Comment (Abacn) [reply .github/workflows/beam_Java_JMH.yml]: added prefix .github/trigger_files

File: .github/workflows/README.md (hunk 2 above)
Comment (Abacn) [reply .github/workflows/README.md]: updated README.md replacing trigger phrase->trigger file, added instructions on how to manually trigger precommit and postcommits

**Review by damccorm — APPROVED** [review]
Thanks!

**Comment by damccorm** [comment]
Oh, we should probably do the same for benchmarks as well

**Comment by Abacn** [comment]
> Oh, we should probably do the same for benchmarks as well

The hope is we can still have the comment trigger in the near future, which is more convenient for developers and reviewers (reviewer can also run the test by comment), as @volatilemolotov working on GitHub App approach. If decided this being long time solution then we can add it to load tests and performance tests

**Pull request merged**